server/hwapi/external/c3/client.py

```diff
@@ -250,4 +280,4 @@ def _load_devices_from_response(
         )
         if created or device not in report.devices:
             report.devices.append(device)
-            self.db.commit()
+        self.db.commit()
```

Comment:
I'm not sure whether I should commit everything at the end or commit separately after changing the codename, and then after modifying the report